webapp/channels/src/components/common/comment_icon.tsx

```diff
@@ -42,23 +48,11 @@ const CommentIcon = ({
         iconStyle = `${iconStyle} ${searchStyle}`;
     }
 
-    const tooltip = (
-        <Tooltip
-            id='comment-icon-tooltip'
-            className='hidden-xs'
-        >
-            <FormattedMessage
-                id='post_info.comment_icon.tooltip.reply'
-                defaultMessage='Reply'
-            />
-        </Tooltip>
-    );
-
     return (
-        <OverlayTrigger
-            delayShow={500}
+        <WithTooltip
+            id='comment-icon-tooltip'
             placement='top'
-            overlay={tooltip}
+            title={messages.replyTooltipTitle}
```

Comment:
similarly for here as well, it already uses `useIntl`

Reply:
For readability sake, I would do as Zubair mention, use here directly `intl.formatMessage`.

But being the devil's advocate... There is a use for this this way. Defining the message outside of the component, it creates only one object for the whole lifetime of the program, adding less stress to the GC. And another nice add, is that the object reference is constant between renders, so it shouldn't cause a re-render in the child component. And the cherry on top is that references are easier to compare than strings.

But I wish that was our bottleneck! For now we can prioritize legibility on this.

The main reasons to use `defineMessages` are to avoid repeating the same id in different parts of the code and to make the code more legible.